pkg/ruler/worker.go

```diff
@@ -0,0 +1,63 @@
+package ruler
+
+import (
+	"time"
+
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/go-kit/kit/log/level"
+)
+
+// Worker does a thing until it's told to stop.
+type Worker interface {
+	Run()
+	Stop()
+}
+
+type worker struct {
+	scheduler *scheduler
+	ruler     *Ruler
+
+	quit chan struct{}
+	done chan struct{}
+}
+
+func newWorker(ruler *Ruler) worker {
+	return worker{
+		scheduler: ruler.scheduler,
+		ruler:     ruler,
+		quit:      make(chan struct{}),
+		done:      make(chan struct{}),
+	}
+}
+
+func (w *worker) Run() {
+	defer close(w.done)
+	for {
+		select {
+		case <-w.quit:
+			return
+		default:
+		}
+		waitStart := time.Now()
+		blockedWorkers.Inc()
```

Comment:
I see these metrics (`blockedWorkers`, `evalLatency`, `workerIdleTime`) are defined in `ruler.go`, can you move them here?

Reply:
👍